airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionReplicationTab.tsx

```diff
@@ -90,12 +90,8 @@ export const ConnectionReplicationTab: React.FC = () => {
       // This could be due to user changes (e.g. in the sync mode) or due to new/removed
       // streams due to a "refreshed source schema".
       const catalogHasChanged = !equal(
-        formValues.syncCatalog.streams
-          .filter((s) => s.config?.selected)
-          .sort(naturalComparatorBy((syncStream) => syncStream.stream?.name ?? "")),
-        connection.syncCatalog.streams
-          .filter((s) => s.config?.selected)
-          .sort(naturalComparatorBy((syncStream) => syncStream.stream?.name ?? ""))
```

Comment:
I'm not sure I understand why these no longer need the sort.  The only place I see us sorting now is the catalogTree.  Isn't there still a chance that the order is not preserved between the connection object and the form values?

Reply:
The reason why this was sorted in the first place was that the formValues data was getting mutated by the sort call in the CatalogTree component. The formValues now contain same stream order as received in the connection object, and the order should match.